ddtrace/profiling/profiler.py

```diff
@@ -364,6 +360,11 @@ def start_collector(collector_class: Type) -> None:
                 before_flush=self._collectors_snapshot,
             )
 
+        # The memory allocation profiler can't be enabled until after the exporters are built, since it optionally
```

Comment:
do we need to do this for stack and lock profilers too?

Reply:
Good call.  Let me refactor this a bit, since our initialization sequence is pretty hairy.